dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/ProgramParser.java

```diff
@@ -71,12 +71,14 @@ public Program parse(String raw, String path, String format, String cflags) thro
 
     private ParserInterface getConcreteParser(File file) throws IOException {
         String name = file.getName();
-        String format = name.substring(name.lastIndexOf(".") + 1);
-        switch (format) {
-            case "ll":
-                return new ParserLlvm();
-            case "litmus":
-                return getConcreteLitmusParser(readFirstLine(file).toUpperCase());
+        if (name.endsWith(".ll")) {
+            return new ParserLlvm();
+        }
+        if (name.endsWith(".spv.dis")) {
```

Comment:
This is just a name I came up with to differentiate between the binary file (`*.spv`) and the assembly one.

AFAIK there is no "standard" extension for spirv assembler so I am ok with keeping this one. However, when we make the support of spriv official, we need to document what is the expected input extension.

BTW ... you also need to add this format to `Dartagnan.supportedFormats` to be able to parse the files from the console

Reply:
I would keep it hidden until we have the actual support, including the verification of spirv files